src/components/activity/ActivityRegistrator.tsx

```diff
@@ -0,0 +1,194 @@
+import React from 'react';
+import { View, ScrollView, KeyboardAvoidingView, TouchableWithoutFeedback, Keyboard, Dimensions, Platform  } from 'react-native';
+import { Text, IconButton, Avatar, useTheme } from 'react-native-paper';
+
+import Slider from '@react-native-community/slider';
+
+import { DatePicker } from '../DatePicker';
+import { TimePicker, getCurrentTimeRounded } from '../TimePicker';
+import { SuggestiveTextInput } from '../SuggestiveTextInput';
+
+import { useTranslation } from 'language/LanguageProvider';
+import Storage from 'storage';
+
+export const ActivityRegistrator = ({ route, navigation }: any) => {
+  // route.params contains information from activity screen
+
+  const lang = useTranslation();
+  const { iconSizes, colors } = useTheme();
+
+  const steps = 1;
+  const defaultChoise = {
+    value: lang.activityRegistratorActivityDefaultChoise,
+    isDefault: true
+  }
+  let choises = [ defaultChoise ];
+
+
+  const [values, modifyValues] = Storage.useValues();
+  const [activities, modifyActivities] = Storage.useActivities();
+  
+  const [fromTime, setFromTime] = React.useState(getCurrentTimeRounded(0, steps));
+  const [toTime, setToTime] = React.useState(getCurrentTimeRounded(1, steps));
+  
+  const [date, setDate] = React.useState(new Date());
+  
+  const [choise, setChoise] = React.useState(defaultChoise);
+  const [activityText, setActivityText] = React.useState('');
+
+  const [importance, setImportance] = React.useState(5); 
+  const [enjoyment, setEnjoyment] = React.useState(5);
+
+
+  // Find topics to choose from depending on input icon
+  const addTopicEntries = (topics: any) => {
+    for(let topicIndex = 0; topicIndex < topics.length; ++topicIndex) {
+      for(let entryIndex = 0; entryIndex < topics[topicIndex].entries.length; ++entryIndex){
+        const entry = topics[topicIndex].entries[entryIndex];
+
+        // If icons match, include it in choises
```

Comment:
```suggestion
        // If icons match, include it in choices
```
Comments must also be updated!